### DISCUSSION TITLE
Step randomization and weight-value?

### DISCUSSION THREAD

**Discussion by routaz** [VIP-Logic/NBomber#662] [Q&A] [answered]
Hello!
I just stumbled across NBomber and it feels very promising at first glance, so big thank you for this!
I've been doing performance testing with Python and Locust.io, but would like to jump ships to NBomber side.
In our tests we have need to randomise steps and set step excution ratio, similiar to how it's done in locust. Is this possible with NBomber?
[https://docs.locust.io/en/stable/writing-a-locustfile.html#task-decorator](
https://docs.locust.io/en/stable/writing-a-locustfile.html#task-decorator)

**Comment by AntyaDev** [comment, marked as answer]
Hi @routaz ,
Thank you for your feedback:) We appreciate it!

You can achieve it on the level of Step but not Scenario.
https://nbomber.com/docs/nbomber/dynamic-workloads#multinomial-distribution

Does it make sense?

**Reply by AntyaDev** [reply]
@routaz 
In your case with Locust - the load simulation configuration (for example, inject(rate: 10)) for, let's say, 5 scenarios will be the same, right?

I am asking to understand how we could model such configuration in NBomber to run several scenarios with execution ratio.

For example
```
NBomberRunner.RegisterScenarios(scn1, scn2, scn3)
             .WithExecutionRatio(0.8, 0.1, 0.1)
```
For such a case, I think all scenarios should have the same Load Simulation configuration.

**Reply by routaz** [reply]
Thank you so much for quick response. I believe being able to give values and randomize execution on step level was exactly what I was looking for! Currently I don't plan to run more than one scenario at a time. I am still in a very beginning of planning our new test pipeline, so the need for more scenarios might come up, but I agree that scenarios should have same Load Simulation configuration. Thank you @AntyaDev

**Reply by AntyaDev** [reply]
Hi @routaz 
We added this feature :)
https://github.com/PragmaticFlow/NBomber/issues/663
